src/main/java/sx/blah/discord/handle/obj/IGuild.java

```diff
@@ -543,6 +543,16 @@
 	IEmoji getEmojiByName(String name);
 
 	/**
+	 * Creates a new emoji.
+	 *
+	 * @param name The name, <b>without colons</b> of length 2-32 characters only consisting of alphanumeric characters and underscores.
+	 * @param image The image of the emoji.
+	 * @param roles The roles for which this emoji will be whitelisted, if empty all roles will be allowed. Your bot must be whitelisted by Discord to use this feature.
```

Comment:
Do bots still need to be whitelisted by Discord? I'm assuming no but I am probably wrong

Reply:
They do to restrict roles. Not to add emojis.

Reply:
Bots need to be whitelisted to use the role whitelist